### ISSUE TITLE
dotnet CLI is 3.0 even though docs says 2.2

### ISSUE THREAD

**Issue by simonauner** [microsoft/azure-pipelines-image-generation#1301]
**Describe the bug**
We're running a build with `ubuntu-latest` that uses `dotnet` CLI, and up until recently that default version has been `2.2.x`. We noticed that sometime during last week this was bumped to `3.0.100`. This breaks our build because the behavior of `dotnet publish --output my-folder` is no longer relative to the project folder, but ends up in the root of the build. Fixable, yes, but it's a bit unexpected to bump the software without notice.

I guess the actual **bug** is that the documentation isn't correct anymore: 
https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops#use-a-microsoft-hosted-agent => https://github.com/Microsoft/azure-pipelines-image-generation/blob/master/images/linux/Ubuntu1604-README.md

**Expected behavior**
I expect the default version of a software (in this case `dotnet` CLI) to be documented in the README above.

**Actual behavior**
`dotnet` CLI is running version `3.0.100` which is undocumented.

![image](https://user-images.githubusercontent.com/1343979/66367512-ccaf8f80-e994-11e9-9507-d40ce563d696.png)

Steps to reproduce:
1. Create a build with YAML file
2. Choose pool `ubuntu-latest`
3. Run a script task: `dotnet --version`

**Image impacted**
 `ubuntu-latest`

**Comment by simonauner** [comment]
This got a bit worse yesterday, when also `windows-latest'` runs .NET Core 3.0. Now the `UseDotNet` task fails to upgrade to 3.x if 3.x is already the default SDK.

**How to replicate:**
1. Run a job with `vmImage: 'windows-latest`
2. Run `UseDotNet` task:
    ```yaml
    - task: UseDotNet@2
      displayName: 'Update .NET Core sdk'
      inputs:
        packageType: sdk
        version: 3.x
    ```

![use-dot-net](https://user-images.githubusercontent.com/1343979/66981038-ef3b5a00-f0b2-11e9-9bd6-147d08486841.png)

**Comment by hiyadav** [comment]
We have updated the installer scripts for both Windows and Linux images, and should start reflecting latest SDK's (in image and documentation) from next release.

**Comment by hiyadav** [comment]
@simonauner the UseDotNet task is failing for another reason, we have a PR to fix this #11577 
This PR shall fix the task for downloading latest 3.0 version.
